components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutionDiagnosticLogger.java

```diff
@@ -0,0 +1,226 @@
+
+import org.apache.http.client.methods.HttpPost;
+import org.wso2.carbon.identity.action.execution.ActionExecutionLogConstants;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.utils.DiagnosticLog;
+
+import java.util.List;
+
+/**
+ * This utility class is for generating and printing diagnostic logs related to
+ * action execution
+ */
+public class ActionExecutionDiagnosticLogger {
+
+    public void printDiagnosticLogActionInitiation(Action action) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        buildDiagnosticLog(
+                setDiagnosticLog(
+                        ActionExecutionLogConstants.ActionIDs.EXECUTE_ACTION,
+                        action.getType().getDisplayName() + " action execution is initiated.",
+                        DiagnosticLog.ResultStatus.SUCCESS));
+    }
+
+    public void printDiagnosticLogActionSkip(ActionType actionType) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        buildDiagnosticLog(
+                setDiagnosticLog(
+                        ActionExecutionLogConstants.ActionIDs.EXECUTE_ACTION,
+                        "Skip executing action for " + actionType + " type.",
+                        DiagnosticLog.ResultStatus.FAILED));
+    }
+
+    public void printDiagnosticLogActionRequest(Action action) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = setDiagnosticLog(
+                ActionExecutionLogConstants.ActionIDs.PROCESS_ACTION_REQUEST,
+                "Call external service endpoint " + action.getEndpoint().getUri() + " for "
+                        + action.getType().getDisplayName() + " action.",
+                DiagnosticLog.ResultStatus.SUCCESS);
+
+        buildDiagnosticLog(addActionConfigParams(diagnosticLogBuilder, action));
+    }
+
+    public void printDiagnosticLogSuccessResponse(Action action) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = setDiagnosticLog(
+                ActionExecutionLogConstants.ActionIDs.RECEIVE_ACTION_RESPONSE,
+                "Received success response from external endpoint " +
+                        action.getEndpoint().getUri() + " for " +
+                        action.getType().getDisplayName() + " action.",
+                DiagnosticLog.ResultStatus.SUCCESS);
+        buildDiagnosticLog(addActionConfigParams(diagnosticLogBuilder, action));
+    }
+
+    public void printDiagnosticLogErrorResponse(Action action) {
```

Comment:
```suggestion
    public void logErrorResponse(Action action) {
```
Fix other places as well

Reply:
Updated in